### ISSUE TITLE
NativeScript 1.6 Problem, please take a look :)

### ISSUE THREAD

**Issue by enchev** [NativeScript/nativescript-cli#1540]
_From @wisonye on February 25, 2016 2:48_

1. Install issue, my NativeScript 1.5 was fine, when I do update or re-install nativescript 1.6, it show this error( should call "tips" ? ), I use "sudo npm install -g nativescript", and actually my $ANDROID_HOME works fine, "$ANDROID_HOME/tools/android" can show the "Android SDK Manager" and all SDK works fine and AVD works fine, do you think that's bug ?

```javascript
> nativescript@1.6.1 postinstall /usr/local/lib/node_modules/nativescript
> node postinstall.js

The ANDROID_HOME environment variable is not set or it points to a non-existent directory. You will not be able to perform any build-related operations for Android.
To be able to perform Android build-related operations, set the ANDROID_HOME variable to point to the root of your Android SDK installation directory.

WARNING: adb from the Android SDK is not installed or is not configured properly.
For Android-related operations, the NativeScript CLI will use a built-in version of adb.
To avoid possible issues with the native Android emulator, Genymotion or connected
Android devices, verify that you have installed the latest Android SDK and
its dependencies as described in http://developer.android.com/sdk/index.html#Requirements

TIP: To avoid setting up the necessary environment variables, you can use the Homebrew package manager to install the Android SDK and its dependencies.

WARNING: The Android SDK is not installed or is not configured properly.
You will not be able to build your projects for Android and run them in the native emulator.
To be able to build for Android and run apps in the native emulator, verify that you have
installed the latest Android SDK and its dependencies as described in http://developer.android.com/sdk/index.html#Requirements

TIP: To avoid setting up the necessary environment variables, you can use the Homebrew package manager to install the Android SDK and its dependencies.

Verifying CocoaPods. This may take more than a minute, please be patient.
npm ERR! registry error parsing json
WARNING: There was a problem with CocoaPods
Verify that CocoaPods are configured properly.
Cannot find a compatible Android SDK for compilation. To be able to build for Android, install Android SDK 22 or later.
Run $ android to manage your Android SDK versions.

You need to have the Android SDK Build-tools installed on your system. You can install any version in the following range: '>=22 <=23'.
Run android from your command-line to install required Android Build Tools. In case you already have them installed, make sure ANDROID_HOME environment variable is set correctly.

You need to have Android SDK 22 or later and the latest Android Support Repository installed on your system.
Run $ android  to manage the Android Support Repository. In case you already have it installed, make sure ANDROID_HOME environment variable is set correctly.
```

and what problem is this ???

```javascript
Verifying CocoaPods. This may take more than a minute, please be patient.
npm ERR! registry error parsing json
WARNING: There was a problem with CocoaPods
```

_Copied from original issue: NativeScript/NativeScript#1640_

**Comment by Plamen5kov** [comment]
Hi @wisonye,
Are you sure that the result of `echo $ANDROID_HOME` is an existing folder pointing to the root of Android SDK and **can be accessed by the user you are currently in**, and the user has permission to execute shell files from that folder.

**Comment by wisonye** [comment]
@Plamen5kov Sure, of course, here is the command and output :)

```javascript
Wisons-MacBook-Pro:NativeScriptProject wison$ echo $ANDROID_HOME
/Users/wison/android-sdk-macosx
Wisons-MacBook-Pro:NativeScriptProject wison$ ll /Users/wison/android-sdk-macosx
total 8
-rw-rw-r--@  1 wison  staff   1.1K 13 Oct 18:58 SDK Readme.txt
drwxrwxr-x@  2 wison  staff    68B 13 Oct 18:58 add-ons
drwxr-xr-x   4 wison  staff   136B 27 Dec 21:02 build-tools
drwxr-xr-x   5 wison  staff   170B 27 Dec 21:03 extras
drwxr-xr-x  13 wison  staff   442B 22 Dec 22:13 platform-tools
drwxrwxr-x@  5 wison  staff   170B 27 Dec 21:09 platforms
drwxr-xr-x   5 wison  staff   170B 27 Dec 21:59 system-images
drwxr-xr-x   2 wison  staff    68B 27 Dec 21:59 temp
drwxr-xr-x  31 wison  staff   1.0K 22 Dec 22:18 tools
```

**Comment by wisonye** [comment]
@Plamen5kov And actually I also try use "sudo" to call the "android", and it's fine, I remember the "--log trace" seems use this command, then I think your install script should use it as well ( to confirm your install script can run the "android", right ? )

```javascript
Wisons-MacBook-Pro:NativeScriptProject wison$ sudo /Users/wison/android-sdk-macosx/tools/android -h

       Usage:
       android [global options] action [action options]
       Global options:
  -s --silent     : Silent mode, shows errors only.
  -v --verbose    : Verbose mode, shows errors, warnings and all messages.
     --clear-cache: Clear the SDK Manager repository manifest cache.
  -h --help       : Help on a specific command.
```

**Comment by Plamen5kov** [comment]
Ok @wisonye, everything looks great so my next question would be can you execute the `adb` command. The adb should be in `/Users/wison/android-sdk-macosx/platform-tools/adb`.
Did you try: https://github.com/NativeScript/nativescript-cli/issues/1425 ?

**Comment by wisonye** [comment]
@Plamen5kov Sure, no problem, run it, result:

```javascript
Wisons-MacBook-Pro:NativeScriptProject wison$ /Users/wison/android-sdk-macosx/platform-tools/adb
Android Debug Bridge version 1.0.32
Revision 09a0d98bebce-android

 -a                            - directs adb to listen on all interfaces for a connection
 -d                            - directs command to the only connected USB device
                                 returns an error if more than one USB device is present.
 -e                            - directs command to the only running emulator.
                                 returns an error if more than one emulator is running.
 -s <specific device>          - directs command to the device or emulator with the given
                                 serial number or qualifier. Overrides ANDROID_SERIAL

........( ignore, too long, result is the help )
```

**Comment by wisonye** [comment]
@Plamen5kov If you're saying I SHOULD use "sudo npm i -g nativescript --unsafe-perm" to install NativeScript, then I didn't try, I try later and then let you know the result, thanks for your good job :) NativeScript is the BEST( not just better )

**Comment by wisonye** [comment]
@Plamen5kov I've tried that, but it still report that warning, here is the output when I install NativeScript via "sudo npm i -g nativescript --unsafe-perm" :

```javascript
> fibers@1.0.6 install /usr/local/lib/node_modules/nativescript/node_modules/fibers
> node build.js || nodejs build.js

`darwin-x64-v8-4.5` exists; testing
Binary is fine; exiting

> utf-8-validate@1.0.1 install /usr/local/lib/node_modules/nativescript/node_modules/utf-8-validate
> node ./build.js

`darwin-x64-v8-4.5` exists; testing
Binary is fine; exiting

> ref@1.1.3 install /usr/local/lib/node_modules/nativescript/node_modules/ref
> node ./build.js

`darwin-x64-v8-4.5` exists; testing
Binary is fine; exiting

> bufferutil@1.0.1 install /usr/local/lib/node_modules/nativescript/node_modules/bufferutil
> node ./build.js

`darwin-x64-v8-4.5` exists; testing
Binary is fine; exiting

> ffi@2.0.0 install /usr/local/lib/node_modules/nativescript/node_modules/ffi
> node ./build.js

`darwin-x64-v8-4.5` exists; testing
Binary is fine; exiting

> ws@0.4.32 install /usr/local/lib/node_modules/nativescript/node_modules/node-inspector/node_modules/ws
> (node-gyp rebuild 2> builderror.log) || (exit 0)

/usr/local/bin/nativescript -> /usr/local/lib/node_modules/nativescript/bin/nativescript.js
/usr/local/bin/tns -> /usr/local/lib/node_modules/nativescript/bin/nativescript.js

> nativescript@1.6.1 postinstall /usr/local/lib/node_modules/nativescript
> node postinstall.js

The ANDROID_HOME environment variable is not set or it points to a non-existent directory. You will not be able to perform any build-related operations for Android.
To be able to perform Android build-related operations, set the ANDROID_HOME variable to point to the root of your Android SDK installation directory.

WARNING: adb from the Android SDK is not installed or is not configured properly.
For Android-related operations, the NativeScript CLI will use a built-in version of adb.
To avoid possible issues with the native Android emulator, Genymotion or connected
Android devices, verify that you have installed the latest Android SDK and
its dependencies as described in http://developer.android.com/sdk/index.html#Requirements

TIP: To avoid setting up the necessary environment variables, you can use the Homebrew package manager to install the Android SDK and its dependencies.

WARNING: The Android SDK is not installed or is not configured properly.
You will not be able to build your projects for Android and run them in the native emulator.
To be able to build for Android and run apps in the native emulator, verify that you have
installed the latest Android SDK and its dependencies as described in http://developer.android.com/sdk/index.html#Requirements

TIP: To avoid setting up the necessary environment variables, you can use the Homebrew package manager to install the Android SDK and its dependencies.

WARNING: CocoaPods is not installed or is not configured properly.
You will not be able to build your projects for iOS if they contain plugin with CocoaPod file.
To be able to build such projects, verify that you have installed CocoaPods.
Cannot find a compatible Android SDK for compilation. To be able to build for Android, install Android SDK 22 or later.
Run $ android to manage your Android SDK versions.

You need to have the Android SDK Build-tools installed on your system. You can install any version in the following range: '>=22 <=23'.
Run android from your command-line to install required Android Build Tools. In case you already have them installed, make sure ANDROID_HOME environment variable is set correctly.

You need to have Android SDK 22 or later and the latest Android Support Repository installed on your system.
Run $ android  to manage the Android Support Repository. In case you already have it installed, make sure ANDROID_HOME environment variable is set correctly.

nativescript@1.6.1 /usr/local/lib/node_modules/nativescript
├── mute-stream@0.0.5
├── bufferpack@0.0.6
├── winreg@0.0.12
├── open@0.0.5
├── properties-parser@0.2.3
├── lockfile@1.0.1
├── xmldom@0.1.21
├── bplist-parser@0.1.0
├── esprima@2.7.0
├── semver@5.0.1
├── marked@0.3.3
├── tabtab@0.0.3
├── filesize@3.1.2
├── colors@1.1.2
├── node-uuid@1.4.3
├── xmlhttprequest@1.5.1
├── osenv@0.1.3 (os-tmpdir@1.0.1, os-homedir@1.0.1)
├── iconv-lite@0.4.11
├── temp@0.8.3 (os-tmpdir@1.0.1, rimraf@2.2.8)
├── byline@4.2.1
├── chalk@1.1.0 (supports-color@2.0.0, escape-string-regexp@1.0.5, strip-ansi@3.0.1, has-ansi@2.0.0, ansi-styles@2.2.0)
├── shelljs@0.5.3
├── minimatch@2.0.10 (brace-expansion@1.1.3)
├── mkdirp@0.5.1 (minimist@0.0.8)
├── bluebird@2.9.34
├── vinyl-filter-since@2.0.2 (readable-stream@2.0.5)
├── through2@2.0.0 (xtend@4.0.1, readable-stream@2.0.5)
├── ws@0.7.1 (options@0.0.6, ultron@1.0.2)
├── rimraf@2.4.2 (glob@5.0.15)
├── inquirer@0.9.0 (strip-ansi@3.0.1, ansi-regex@2.0.0, figures@1.4.0, rx-lite@2.5.2, through@2.3.8, cli-width@1.1.1, readline2@0.1.1, run-async@0.1.0)
├── plist-merge-patch@0.0.9 (json-merge-patch@0.2.3, plist@1.2.0)
├── progress-stream@1.1.1 (single-line-log@0.3.1, speedometer@0.1.4, through2@0.2.3)
├── gaze@0.5.2 (globule@0.1.0)
├── xcode@0.8.0 (node-uuid@1.3.3, pegjs@0.6.2)
├── fibers@1.0.6
├── moment@2.10.6
├── utf-8-validate@1.0.1 (bindings@1.2.1, prebuilt@0.1.0, nan@2.2.0)
├── ref@1.1.3 (bindings@1.2.1, prebuilt@0.1.0, debug@2.2.0, nan@2.2.0)
├── plistlib@0.2.1 (underscore@1.5.2, node-xml@1.0.2, async@0.2.10, xml@0.0.12, moment@2.4.0)
├── bufferutil@1.0.1 (bindings@1.2.1, prebuilt@0.1.0, nan@2.2.0)
├── broccoli@0.16.4 (broccoli-slow-trees@1.1.0, copy-dereference@1.0.0, promise-map-series@0.2.2, mime@1.3.4, commander@2.9.0, rsvp@3.2.1, findup-sync@0.2.1, broccoli-kitchen-sink-helpers@0.2.9, quick-temp@0.1.5, connect@3.4.1, handlebars@3.0.3)
├── log4js@0.6.26 (async@0.2.10, underscore@1.8.2, semver@4.3.6, readable-stream@1.0.33)
├── lodash@3.10.0
├── cli-table@0.3.1 (colors@1.0.3, wcwidth@1.0.0, lodash@3.6.0)
├── broccoli-funnel@0.2.3 (walk-sync@0.1.3, symlink-or-copy@1.0.1, broccoli-read-compat@0.1.3, core-object@0.0.2)
├── plist@1.1.0 (util-deprecate@1.0.0, base64-js@0.0.6, xmlbuilder@2.2.1)
├── ref-struct@1.0.2 (debug@2.2.0)
├── clui@0.3.1 (cli-color@0.3.2)
├── ffi@2.0.0 (bindings@1.2.1, prebuilt@0.1.0, debug@2.2.0, nan@2.2.0)
├── yargs@3.15.0 (camelcase@1.2.1, window-size@0.1.4, decamelize@1.1.2, cliui@2.1.0)
├── marked-terminal@1.5.0 (node-emoji@0.1.0, lodash.assign@3.2.0, cardinal@0.5.0)
├── npm@2.14.12
├── gulp@3.9.0 (interpret@0.6.6, pretty-hrtime@1.0.2, deprecated@0.0.1, archy@1.0.0, tildify@1.1.2, minimist@1.2.0, semver@4.3.6, v8flags@2.0.11, orchestrator@0.3.7, liftoff@2.2.0, vinyl-fs@0.3.14, gulp-util@3.0.7)
├── node-inspector@0.7.4 (debug@0.8.1, opener@1.3.0, which@1.0.9, async@0.8.0, rc@0.3.5, glob@3.2.11, yargs@1.2.6, strong-data-uri@0.1.1, ws@0.4.32, express@4.0.0)
└── ios-sim-portable@1.0.20 (colors@0.6.2, lodash@3.2.0, nodobjc@2.0.0)
^C
Wisons-MacBook-Pro:temp wison$ vi ./install.log 

├── xmldom@0.1.21
├── bplist-parser@0.1.0
├── esprima@2.7.0
├── semver@5.0.1
├── marked@0.3.3
├── tabtab@0.0.3
├── filesize@3.1.2
├── colors@1.1.2
├── node-uuid@1.4.3
├── xmlhttprequest@1.5.1
├── osenv@0.1.3 (os-tmpdir@1.0.1, os-homedir@1.0.1)
├── iconv-lite@0.4.11
├── temp@0.8.3 (os-tmpdir@1.0.1, rimraf@2.2.8)
├── byline@4.2.1
├── chalk@1.1.0 (supports-color@2.0.0, escape-string-regexp@1.0.5, strip-ansi@3.0.1, has-ansi@2.0.0, ansi-styles@2.2.0)
├── shelljs@0.5.3
├── minimatch@2.0.10 (brace-expansion@1.1.3)
├── mkdirp@0.5.1 (minimist@0.0.8)
├── bluebird@2.9.34
├── vinyl-filter-since@2.0.2 (readable-stream@2.0.5)
├── through2@2.0.0 (xtend@4.0.1, readable-stream@2.0.5)
├── ws@0.7.1 (options@0.0.6, ultron@1.0.2)
├── rimraf@2.4.2 (glob@5.0.15)
├── inquirer@0.9.0 (strip-ansi@3.0.1, ansi-regex@2.0.0, figures@1.4.0, rx-lite@2.5.2, through@2.3.8, cli-width@1.1.1, readline2@0.1.1, run-async@0.1.0)
├── plist-merge-patch@0.0.9 (json-merge-patch@0.2.3, plist@1.2.0)
├── progress-stream@1.1.1 (single-line-log@0.3.1, speedometer@0.1.4, through2@0.2.3)
├── gaze@0.5.2 (globule@0.1.0)
├── xcode@0.8.0 (node-uuid@1.3.3, pegjs@0.6.2)
├── fibers@1.0.6
├── moment@2.10.6
├── utf-8-validate@1.0.1 (bindings@1.2.1, prebuilt@0.1.0, nan@2.2.0)
├── ref@1.1.3 (bindings@1.2.1, prebuilt@0.1.0, debug@2.2.0, nan@2.2.0)
├── plistlib@0.2.1 (underscore@1.5.2, node-xml@1.0.2, async@0.2.10, xml@0.0.12, moment@2.4.0)
├── bufferutil@1.0.1 (bindings@1.2.1, prebuilt@0.1.0, nan@2.2.0)
├── broccoli@0.16.4 (broccoli-slow-trees@1.1.0, copy-dereference@1.0.0, promise-map-series@0.2.2, mime@1.3.4, commander@2.9.0, rsvp@3.2.1, findup-sync@0.2.1, broccoli-kitchen-sink-helpers@0.2.9, quick-temp@0.1.5, connect@3.4.1, handlebars@3.0.3)
├── log4js@0.6.26 (async@0.2.10, underscore@1.8.2, semver@4.3.6, readable-stream@1.0.33)
├── lodash@3.10.0
├── cli-table@0.3.1 (colors@1.0.3, wcwidth@1.0.0, lodash@3.6.0)
├── broccoli-funnel@0.2.3 (walk-sync@0.1.3, symlink-or-copy@1.0.1, broccoli-read-compat@0.1.3, core-object@0.0.2)
├── plist@1.1.0 (util-deprecate@1.0.0, base64-js@0.0.6, xmlbuilder@2.2.1)
├── ref-struct@1.0.2 (debug@2.2.0)
├── clui@0.3.1 (cli-color@0.3.2)
├── ffi@2.0.0 (bindings@1.2.1, prebuilt@0.1.0, debug@2.2.0, nan@2.2.0)
├── yargs@3.15.0 (camelcase@1.2.1, window-size@0.1.4, decamelize@1.1.2, cliui@2.1.0)
├── marked-terminal@1.5.0 (node-emoji@0.1.0, lodash.assign@3.2.0, cardinal@0.5.0)
├── npm@2.14.12
├── gulp@3.9.0 (interpret@0.6.6, pretty-hrtime@1.0.2, deprecated@0.0.1, archy@1.0.0, tildify@1.1.2, minimist@1.2.0, semver@4.3.6, v8flags@2.0.11, orchestrator@0.3.7, liftoff@2.2.0, vinyl-fs@0.3.14, gulp-util@3.0.7)
├── node-inspector@0.7.4 (debug@0.8.1, opener@1.3.0, which@1.0.9, async@0.8.0, rc@0.3.5, glob@3.2.11, yargs@1.2.6, strong-data-uri@0.1.1, ws@0.4.32, express@4.0.0)
└── ios-sim-portable@1.0.20 (colors@0.6.2, lodash@3.2.0, nodobjc@2.0.0)
```

**Comment by teobugslayer** [comment]
Hi @wisonye,

Can you try this. Start node (on the console type node). When you are in the node REPL, type
```
process.env.ANDROID_HOME
```

This should print the value of ANDROID_HOME as seen by NativeScipt CLI as opposed as seen by your shell. Sometimes they can differ in more complex scenarios. Check if the value printed is correct.

You mentioned another problem - with CocoaPods. Currently I do not have access to a mac to check, but on top of my head, can you type in Terminal

```
pod --version
```

It should say something similar to 0.39.

**Comment by wisonye** [comment]
@teobugslayer Here is my print out which you ask for:

```javascript
Wisons-MacBook-Pro:NativeScriptProject wison$ echo $ANDROID_HOME 
/Users/wison/android-sdk-macosx
Wisons-MacBook-Pro:NativeScriptProject wison$ node
> process.env.ANDROID_HOME
'/Users/wison/android-sdk-macosx'
> 
```

And here is pod version:

```javascript
Wisons-MacBook-Pro:NativeScriptProject wison$ pod --version
0.39.0
```

**Comment by rosen-vladimirov** [comment]
Hi @wisonye , 
May I ask you for your version of MacOS. It looks like Yosemite has some issues with Environment variables. Maybe that's the case on your side?

**Comment by wisonye** [comment]
@rosen-vladimirov My Yosemite version is  "10.10.5", and all my environment setup in my "~/.bash_profile" :)

**Comment by enchev** [comment]
Hey @wisonye,

Do you still have problem with this using our latest version?

**Comment by fornsss** [comment]
hi I have a problem I instaling in my windows nativescript but I have a problem the ANDROID_HOME 
environment variable is not set or it points ...., and I don't now what is wrong 


![1](https://cloud.githubusercontent.com/assets/4648701/24366293/f8d431a6-1318-11e7-95c4-c45182b48d19.png)

**Comment by Plamen5kov** [comment]
Hi @fornsss the `ANDROID_HOME` environment variable should point to the root folder of the  android SDK.

**Comment by mkoseoglu** [comment]
@wisonye @enchev I have this problem, too. How did you solve it? I use Linux.

**Comment by fuser84** [comment]
Guys,
here is my fix on Ubuntu 16.04.3 LTS. 
Add the below to the .bash_profile(create if you don't have it):

export JAVA_HOME=/usr/lib/jvm/java-8-oracle/
export ANDROID_HOME=/home/(your user)/Android/Sdk
export PATH=${PATH}:/home/(your user)/Android/Sdk/platform-tools:/home/(your user)/Android/Sdk/tools:/home/(your user)/Android/Sdk/tools/bin

at the end ==> $ source ~/.bash_profile

**Comment by b-and-p** [comment]
@fuser84 Excellent, thanks! your step 1 in `~/.bashsrc` did it for me on Linux Mint 17.3
